### DISCUSSION TITLE
SkyWalking Ruby Agent released 0.1.0

### DISCUSSION THREAD

**Discussion by CodePrometheus** [apache/skywalking#12729] [Ideas]
# SkyWalking Ruby Agent Released

SkyWalking Ruby is an open-source auto-instrument agent that provides support for distributed tracing across different frameworks within the Ruby language.

- Source code, https://github.com/apache/skywalking-ruby
- Doc, https://skywalking.apache.org/docs/skywalking-ruby/latest/readme/

___
_Original Post_

## [Ruby Agent Proposal] Exploration and practice of Ruby in the skywalking ecosystem

The plan is still being considered, researched and implemented, and I will update it gradually later.

https://docs.google.com/document/d/1JTpK95m8H5OQNPyF_4VqwO-K_ygp-5WVYOuYUFkJxwc

**Comment by wu-sheng** [comment]
It is good to have this proposal. I would like to suggest, do you consider supporting adding a new field to the target class. Java agent has this mechanism to cache values, such as connection metadata to help on build span and span's peer.

**Reply by CodePrometheus** [reply]
`DynamicField` I'm exploring whether Ruby could achieve this.

**Reply by wu-sheng** [reply]
How is this feature? As it is very performance relative, I hope we have a conclusion for the pre-release.

**Reply by CodePrometheus** [reply]
I add it in [enhance instance](https://docs.google.com/document/d/1JTpK95m8H5OQNPyF_4VqwO-K_ygp-5WVYOuYUFkJxwc/edit?tab=t.0#heading=h.pdclwvr1az1r), it will add a new field to the target class, and generat setter and getter methods.

**Comment by wu-sheng** [comment]
I there any update about this new agent.

**Reply by CodePrometheus** [reply]
Currently the following features from the 0.1 pre-release plan have been completed:
1.Implemented agent startup, configuration, and plugin-loading.
2.Initially completed the scheduling tool, with interface-level implementations of report_instance_properties and keep_alive.
3.Finished most of the tracing-related functionalities, including Span, TraceContext, etc.

Currently in progress:
1.Continuing to complete tracing and report traceSegment to OAP.
2.Setting up e2e testing and completing GitHub Actions.

**Reply by wu-sheng** [reply]
It looks like we are better to start asking for establish new sub-project and sub-repo? Could you write a proposal to the mailing list?

**Reply by CodePrometheus** [reply]
Writing an email is fine, I would like to know this application is for permission to establish a repo or for actually creating a new repo, I mean, could an existing repo [cp/skywalking-ruby](https://github.com/CodePrometheus/skywalking-ruby) be transferred to Apache?

**Reply by wu-sheng** [reply]
Transferring mean donation, that requires IP clearance process.
As you are targeting this to upstream directly, I would say your local part is a draft, and we establish a new repo.

**Reply by CodePrometheus** [reply]
OK, no problem.

**Comment by CodePrometheus** [comment]
I'm verifying the core links these days, and will send out the proposal as soon as possible.

<img width="1425" alt="image" src="https://github.com/user-attachments/assets/aaa0da8f-ad52-4210-beab-8cd9a28e9cda" />

**Reply by wu-sheng** [reply]
We will have time to add plugin tests like other agents, before the 1st release(0.1). Then keep polishing that.

**Reply by CodePrometheus** [reply]
OK, that sounds good.

**Comment by wu-sheng** [comment]
The repo has been created, https://github.com/apache/skywalking-ruby
